src/masternode.cpp

```diff
@@ -17,91 +15,35 @@
 
 
 CMasternode::CMasternode() :
-    vin(),
-    addr(),
-    pubKeyCollateralAddress(),
-    pubKeyMasternode(),
-    lastPing(),
-    vchSig(),
-    sigTime(GetAdjustedTime()),
-    nLastDsq(0),
-    nTimeLastChecked(0),
-    nTimeLastPaid(0),
-    nTimeLastWatchdogVote(0),
-    nActiveState(MASTERNODE_ENABLED),
-    nCacheCollateralBlock(0),
-    nBlockLastPaid(0),
-    nProtocolVersion(PROTOCOL_VERSION),
-    nPoSeBanScore(0),
-    nPoSeBanHeight(0),
-    fAllowMixingTx(true),
-    fUnitTest(false)
+    masternode_info_t{ MASTERNODE_ENABLED, PROTOCOL_VERSION, GetAdjustedTime()},
```

Comment:
Agree - as noted in the commit message, there are pros and cons here.
The existing explicit lists were long and separated from the data itself.
This duplication had already introduced some initialisation errors.

The (pseudo) aggregate initialization is very open to future error but that can be mitigated.
Stronger typing  - if the data member fields had proper types then they could not be initialised out of order - the `int` constants should be `enum` types and we should use a time specific type rather than `uint64_t`. I can make those changes, though it will involve touching more files